packages/builders/src/Assertions.ts

```diff
@@ -1,18 +1,16 @@
 import { Locale } from 'discord-api-types/v10';
-import { z } from 'zod';
+import { z } from 'zod/v4';
 
 export const customIdPredicate = z.string().min(1).max(100);
 
 export const memberPermissionsPredicate = z.coerce.bigint();
 
-export const localeMapPredicate = z
-	.object(
-		Object.fromEntries(Object.values(Locale).map((loc) => [loc, z.string().optional()])) as Record<
-			Locale,
-			z.ZodOptional<z.ZodString>
-		>,
-	)
-	.strict();
+export const localeMapPredicate = z.strictObject(
+	Object.fromEntries(Object.values(Locale).map((loc) => [loc, z.string().optional()])) as Record<
+		Locale,
+		z.ZodOptional<z.ZodString>
+	>,
```

Comment:
Do we really need the cast or can we just cast the returned array `as const` and achieve the same thing?

Reply:
Yes, we do need the cast. This is because the return type of `Object.fromEntries()` disregards the type of the keys.
https://github.com/microsoft/TypeScript/blob/51dcd9001d9dceb98d248f574f39f57497a44968/src/lib/es2019.object.d.ts#L8